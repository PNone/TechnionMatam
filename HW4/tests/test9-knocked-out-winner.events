Slime
Balrog
Slime
Snail
Snail
Snail
Snail
Snail
Snail
Snail
PotionsMerchant