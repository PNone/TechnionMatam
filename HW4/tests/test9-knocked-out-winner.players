Braa Warrior RiskTaking
Regev Magician Responsible