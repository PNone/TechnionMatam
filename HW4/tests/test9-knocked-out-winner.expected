Game is about to start!
These are the players:

1. Braa, Warrior with RiskTaking character (level 1, force 5)
2. Regev, Magician with Responsible character (level 1, force 5)

----------------------------------------

New round starts!

Turn: 1
Player: Braa, Warrior with RiskTaking character (level 1, force 5)
Event: Slime (power 12, loot 5, damage 25)
Braa lost the encounter and took 25 damage!

Turn: 2
Player: Regev, Magician with Responsible character (level 1, force 5)
Event: Balrog (power 15, loot 100, damage 9001)
Regev lost the encounter and took 9001 damage!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            1         5     115   10
2          Regev           1         5     0     10

----------------------------------------

New round starts!

Turn: 3
Player: Braa, Warrior with RiskTaking character (level 1, force 5)
Event: Slime (power 12, loot 5, damage 25)
Braa lost the encounter and took 25 damage!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            1         5     80    10
2          Regev           1         5     0     10

----------------------------------------

New round starts!

Turn: 4
Player: Braa, Warrior with RiskTaking character (level 1, force 5)
Event: Snail (power 5, loot 2, damage 10)
Braa won the encounter, gained 2 coins and leveled up!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            2         5     70    12
2          Regev           1         5     0     10

----------------------------------------

New round starts!

Turn: 5
Player: Braa, Warrior with RiskTaking character (level 2, force 5)
Event: Snail (power 5, loot 2, damage 10)
Braa won the encounter, gained 2 coins and leveled up!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            3         5     60    14
2          Regev           1         5     0     10

----------------------------------------

New round starts!

Turn: 6
Player: Braa, Warrior with RiskTaking character (level 3, force 5)
Event: Snail (power 5, loot 2, damage 10)
Braa won the encounter, gained 2 coins and leveled up!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            4         5     50    16
2          Regev           1         5     0     10

----------------------------------------

New round starts!

Turn: 7
Player: Braa, Warrior with RiskTaking character (level 4, force 5)
Event: Snail (power 5, loot 2, damage 10)
Braa won the encounter, gained 2 coins and leveled up!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            5         5     40    18
2          Regev           1         5     0     10

----------------------------------------

New round starts!

Turn: 8
Player: Braa, Warrior with RiskTaking character (level 5, force 5)
Event: Snail (power 5, loot 2, damage 10)
Braa won the encounter, gained 2 coins and leveled up!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            6         5     30    20
2          Regev           1         5     0     10

----------------------------------------

New round starts!

Turn: 9
Player: Braa, Warrior with RiskTaking character (level 6, force 5)
Event: Snail (power 5, loot 2, damage 10)
Braa won the encounter, gained 2 coins and leveled up!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            7         5     20    22
2          Regev           1         5     0     10

----------------------------------------

New round starts!

Turn: 10
Player: Braa, Warrior with RiskTaking character (level 7, force 5)
Event: Snail (power 5, loot 2, damage 10)
Braa won the encounter, gained 2 coins and leveled up!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            8         5     10    24
2          Regev           1         5     0     10

----------------------------------------

New round starts!

Turn: 11
Player: Braa, Warrior with RiskTaking character (level 8, force 5)
Event: PotionsMerchant
Braa bought 1 potions!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            8         5     20    19
2          Regev           1         5     0     10

----------------------------------------

New round starts!

Turn: 12
Player: Braa, Warrior with RiskTaking character (level 8, force 5)
Event: Slime (power 12, loot 5, damage 25)
Braa won the encounter, gained 5 coins and leveled up!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            9         5     10    24
2          Regev           1         5     0     10

----------------------------------------

New round starts!

Turn: 13
Player: Braa, Warrior with RiskTaking character (level 9, force 5)
Event: Balrog (power 17, loot 100, damage 9001)
Braa won the encounter, gained 100 coins and leveled up!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            10        5     0     124
2          Regev           1         5     0     10

----------------------------------------

Game is over!
The winner is: Braa
Congratulations!

